### PR TITLE
[Issue-3583] Unified account - Some UI bugs for Send TON

### PR DESCRIPTION
**Pull request by dungnguyen-art**
Related issue: #3583

### CODE REVIEW

File: packages/extension-base/src/services/history-service/subscan-history.ts [hunk 1]
```diff
@@ -33,13 +33,13 @@ export function parseSubscanExtrinsicData (address: string, extrinsicItem: Extri
...
   const nativeSymbol = chainInfo.substrateInfo?.symbol || chainInfo.evmInfo?.symbol || '';
-
+  const networkPrefix = chainInfo.substrateInfo?.addressPrefix ?? 42;
   const initData: TransactionHistoryItem = {
     address,
     origin: 'subscan',
     time: extrinsicItem.block_timestamp * 1000,
     chainType,
-    from: address,
+    from: extrinsicItem.account_display.address,
```
Comment (nampc1) [inline packages/extension-base/src/services/history-service/subscan-history.ts:42]: what format is this in?

File: packages/extension-base/src/services/history-service/subscan-history.ts [hunk 2]
```diff
@@ -33,13 +33,13 @@ export function parseSubscanExtrinsicData (address: string, extrinsicItem: Extri
   const chainType = chainInfo.substrateInfo ? ChainType.SUBSTRATE : ChainType.EVM;
   const nativeDecimals = chainInfo.substrateInfo?.decimals || chainInfo.evmInfo?.decimals || 18;
   const nativeSymbol = chainInfo.substrateInfo?.symbol || chainInfo.evmInfo?.symbol || '';
-
+  const networkPrefix = chainInfo.substrateInfo?.addressPrefix ?? 42;
```
Comment (nampc1) [inline packages/extension-base/src/services/history-service/subscan-history.ts:36]: use util instead of direct access

**Review by nampc1 — CHANGES_REQUESTED** [review]

File: packages/extension-base/src/services/history-service/subscan-history.ts [hunk 3]
```diff
@@ -90,14 +90,14 @@ export function parseSubscanTransferData (address: string, transferItem: Transfe
     origin: 'subscan',
     time: transferItem.block_timestamp * 1000,
     chainType,
-    from,
+    from: transferItem.from,
     fromName: transferItem.from_account_display.display || transferItem.from_account_display.address,
     direction: isSameAddress(address, from) ? TransactionDirection.SEND : TransactionDirection.RECEIVED,
```
Comment (S2kael) [inline packages/extension-base/src/services/history-service/subscan-history.ts:95]: Change `from` to `transferItem.from`

File: packages/extension-koni-ui/src/components/Field/Earning/EarningPoolSelector.tsx [hunk 4]
```diff
@@ -408,8 +407,8 @@ const Component = (props: Props, ref: ForwardedRef<InputRef>) => {
         onSelect={_onSelectItem}
         placeholder={placeholder || t('Select pool')}
         prefix={(
-          <Avatar
-            identPrefix={networkPrefix}
+          <AccountProxyAvatar
```
Comment (S2kael) [inline packages/extension-koni-ui/src/components/Field/Earning/EarningPoolSelector.tsx:410]: Confirm this change with BA team

**Review by S2kael — CHANGES_REQUESTED** [review]

File: packages/extension-base/src/services/history-service/subscan-history.ts (hunk 1 above)
Comment (dungnguyen-art) [reply packages/extension-base/src/services/history-service/subscan-history.ts:42]: it is the format of the network address

File: packages/extension-koni-ui/src/components/MetaInfo/parts/AccountItem.tsx [hunk 5]
```diff
@@ -25,22 +22,14 @@ export interface AccountInfoItem extends InfoItemBase {
...
   const account = useGetAccountByAddress(accountAddress);
 
-  const shortAddress = useMemo(() => {
-    if (!chainSlug) {
-      return toShort(accountAddress);
-    }
-
-    const chainInfo = chainSlug ? chainInfoMap[chainSlug] : undefined;
-    const formattedAddress = chainInfo ? _reformatAddressWithChain(accountAddress, chainInfo) : accountAddress;
+  console.log('account', account);
```
Comment (lw-cdm) [inline packages/extension-koni-ui/src/components/MetaInfo/parts/AccountItem.tsx:30]: remove comment

File: packages/extension-base/src/utils/account/common.ts [hunk 6]
```diff
@@ -3,6 +3,7 @@
 
 import { _ChainInfo } from '@subwallet/chain-list/types';
 import { ALL_ACCOUNT_KEY } from '@subwallet/extension-base/constants';
+import { tonAddressInfo } from '@subwallet/extension-base/services/balance-service/helpers/subscribe/ton/utils';
```
Comment (S2kael) [inline packages/extension-base/src/utils/account/common.ts:6]: Remove this line

**Review by S2kael — APPROVED** [review]

File: packages/extension-koni-ui/src/components/Field/AddressInput.tsx [hunk 7]
```diff
@@ -4,6 +4,7 @@
 import { CHAINS_SUPPORTED_DOMAIN, isAzeroDomain } from '@subwallet/extension-base/koni/api/dotsama/domain';
 import { AbstractAddressJson } from '@subwallet/extension-base/types';
 import { reformatAddress } from '@subwallet/extension-base/utils';
+import { AccountProxyAvatar } from '@subwallet/extension-koni-ui/components';
```
Comment (lw-cdm) [inline packages/extension-koni-ui/src/components/Field/AddressInput.tsx:7]: Revert this file

File: packages/extension-koni-ui/src/components/MetaInfo/parts/TransferItem.tsx [hunk 8]
```diff
@@ -62,19 +53,20 @@ const Component: React.FC<TransferInfoItem> = (props: TransferInfoItem) => {
     return '';
   }, [recipientName, senderName]);
 
-  const genAccountBlock = (address: string, name?: string, chainInfo?: _ChainInfo) => {
-    const formattedAddress = chainInfo ? _reformatAddressWithChain(address, chainInfo) : address;
-    const shortAddress = toShort(formattedAddress);
+  const genAccountBlock = (address: string, name?: string) => {
+    const originAddress = reformatAddress(address, 42);
```
Comment (lw-cdm) [inline packages/extension-koni-ui/src/components/MetaInfo/parts/TransferItem.tsx:57]: no need reformat

File: packages/extension-koni-ui/src/Popup/Transaction/variants/SendFund.tsx [hunk 9]
```diff
@@ -403,14 +399,9 @@ const Component = ({ className = '', isAllAccount, targetAccountProxy }: Compone
   }, [accounts, assetRegistry, notification, t]);
 
   const handleBasicSubmit = useCallback((values: TransferParams, options: TransferOptions): Promise<SWTransactionResponse> => {
-    const { asset, chain, destChain, from: _from, to, value } = values;
-
+    const { asset, chain, destChain, from, to, value } = values;
     let sendPromise: Promise<SWTransactionResponse>;
```
Comment (lw-cdm) [inline packages/extension-koni-ui/src/Popup/Transaction/variants/SendFund.tsx:404]: recheck this logic

File: packages/extension-koni-ui/src/components/Modal/AddressBook/AddressBookModal.tsx [hunk 10]
```diff
@@ -168,7 +168,7 @@ const Component: React.FC<Props> = (props: Props) => {
     return (
       <AddressSelectorItem
         address={item.formatedAddress}
-        avatarValue={item.proxyId}
+        avatarValue={item.proxyId || item.formatedAddress}
```
Comment (lw-cdm) [inline packages/extension-koni-ui/src/components/Modal/AddressBook/AddressBookModal.tsx:171]: change to item.address

**Pull request merged**